tensorflow_datasets/scripts/generate_visualization.py

```diff
@@ -23,40 +23,82 @@
 
 import os
 from absl import app
+from absl import flags
+from concurrent import futures
 
 import tensorflow as tf
 import tensorflow_datasets as tfds
+from tensorflow_datasets.core import registered
 
+WORKER_COUNT_DATASETS = 200
+
+FLAGS = flags.FLAGS
 FIG_DIR = os.path.join('..', 'docs', 'catalog', 'images')
-FULL_PATH = tfds.core.get_tfds_path(FIG_DIR)
+flags.DEFINE_string('dst_dir', tfds.core.get_tfds_path(FIG_DIR),
+                    'Path to tensorflow_datasets/docs/catalog/images')
+
+def generate_single_visualization(data_name):
+    """Save the generated figures for the dataset in dst_dir.
 
-def generate_vizz(data_name):
-    """Save the generated figures for the dataset
     Args:
-      data_name: name of the dataset
+      data_name: name of the dataset with or without config for which
+      generate figure.
     """
+    print("Generating examples %s..." % data_name)
+    builder = tfds.builder(data_name)
```

Comment:
Note: If the image already exists, we should skip the generation entirely (to avoid recomputing the same image twice.

Reply:
Actually I include this here only because of next line to it ie, `split = list(builder.info.splits.keys())[0]` as splits for different datasets are different

Reply:
I just meant to add somethng like:

```python
if tf.io.gfile.exists(dst_path):
  return  # If the image already exists, skip the image generation
builder = tfds.builder(data_name)
```

Reply:
okay this is good to include I will make changes